### PR TITLE
Chore: sushi integration tests for all engines

### PR DESCRIPTION
**Pull request by treysp**

### CODE REVIEW

File: examples/sushi/models/customer_revenue_by_day.sql [hunk 1]
```diff
@@ -2,39 +2,39 @@
 MODEL (
   name sushi.customer_revenue_by_day,
   kind incremental_by_time_range (
-    time_column (ds, 'YYYY-MM-dd'),
+    time_column date,
```
Comment (tobymao) [inline examples/sushi/models/customer_revenue_by_day.sql:5]: event_date maybe

File: examples/sushi/models/customer_revenue_by_day.sql [hunk 2]
```diff
@@ -2,39 +2,39 @@
...
     oi.order_id AS order_id,
     SUM(oi.quantity * i.price) AS total,
-    oi.ds AS ds
+    oi.date AS date
   FROM sushi.order_items AS oi
   LEFT JOIN sushi.items AS i
-    ON oi.item_id = i.id AND oi.ds = i.ds
+    ON oi.item_id = i.id AND oi.date = i.date
   WHERE
-    oi.ds BETWEEN '{{ start_ds }}' AND '{{ end_ds }}'
```
Comment (tobymao) [inline examples/sushi/models/customer_revenue_by_day.sql:24]: why did you change this

File: examples/sushi/models/customer_revenue_by_day.sql [hunk 3]
```diff
@@ -2,39 +2,39 @@
...
   SUM(ot.total)::DOUBLE AS revenue, /* Revenue from orders made by this customer */
-  o.ds::TEXT AS ds /* Date */
+  o.date::DATE AS date /* Date */
 FROM sushi.orders AS o
 LEFT JOIN order_total AS ot
-  ON o.id = ot.order_id AND o.ds = ot.ds
+  ON o.id = ot.order_id AND o.date = ot.date
 WHERE
-  o.ds BETWEEN '{{ start_ds }}' AND '{{ end_ds }}'
+  o.date BETWEEN @start_date AND @end_date
```
Comment (tobymao) [inline examples/sushi/models/customer_revenue_by_day.sql:37]: this was intentional

File: tests/core/engine_adapter/test_integration.py [hunk 4]
```diff
@@ -736,12 +743,14 @@ def test_merge(ctx: TestContext):
...
 def test_scd_type_2(ctx: TestContext):
+    time_type = "datetime" if ctx.dialect == "bigquery" else "timestamp"
+
     ctx.columns_to_types = {
         "id": "int",
         "name": "string",
-        "updated_at": "timestamp",
-        "valid_from": "timestamp",
-        "valid_to": "timestamp",
+        "updated_at": f"{time_type}",
```
Comment (tobymao) [inline tests/core/engine_adapter/test_integration.py:751]: these f strings are unnecessary

File: examples/sushi/models/customer_revenue_by_day.sql (hunk 2 above)
Comment (treysp) [reply examples/sushi/models/customer_revenue_by_day.sql:24]: They need to be changed from date strings to dates, and `{{ start_date }}` errored out.

I'll run it again and grab the error.

Comment (tobymao) [reply examples/sushi/models/customer_revenue_by_day.sql:24]: you can make it CAST('{{start_ds}}' as DATE)

Comment (treysp) [reply examples/sushi/models/customer_revenue_by_day.sql:24]: ok - one issue is that it needed the `JINJA_QUERY_BEGIN`, but with that in place duckdb doesn't render `{{ start_date }}` correctly: 

"`oi`.`event_date` <= 2020 AND `oi`.`event_date` >= 2020"

```
2023-11-29 17:39:24,838 - MainThread - sqlmesh.core.model.definition - DEBUG - Rendered query for 'sushi.customer_revenue_by_day':
  WITH `order_total` AS (
    SELECT
      `oi`.`order_id` AS `order_id`,
      SUM(`oi`.`quantity` * `i`.`price`) AS `total`,
      `oi`.`event_date` AS `event_date`
    FROM `sushi`.`order_items__fixture` AS `oi`
    LEFT JOIN `sushi`.`items__fixture` AS `i`
      ON `oi`.`event_date` = `i`.`event_date` AND `oi`.`item_id` = `i`.`id`
    WHERE
      `oi`.`event_date` <= 2020 AND `oi`.`event_date` >= 2020
    GROUP BY
      `oi`.`order_id`,
      `oi`.`event_date`
  )
  SELECT
    CAST(`o`.`customer_id` AS INT) AS `customer_id`, /* Customer id */
    CAST(SUM(`ot`.`total`) AS DOUBLE) AS `revenue`, /* Revenue from orders made by this customer */
    CAST(`o`.`event_date` AS DATE) AS `event_date` /* Date */
  FROM `sushi`.`orders__fixture` AS `o`
  LEFT JOIN `order_total` AS `ot`
    ON `o`.`event_date` = `ot`.`event_date` AND `o`.`id` = `ot`.`order_id`
  WHERE
    `o`.`event_date` <= 2020 AND `o`.`event_date` >= 2020
  GROUP BY
    `o`.`customer_id`,
    `o`.`event_date` (definition.py:986)
```

File: tests/core/engine_adapter/test_integration.py [hunk 5]
```diff
@@ -872,3 +881,74 @@ def test_scd_type_2(ctx: TestContext):
...
+    # clean up schemas
+    for schema in [
+        "sushi__test_prod",
+        "sushi__test_dev",
+        "sushi",
+        "sqlmesh__sushi",
+        "sqlmesh",
+        "raw",
+    ]:
+        context.engine_adapter.drop_schema(schema, ignore_if_not_exists=True, cascade=True)
```
Comment (eakmanrq) [inline tests/core/engine_adapter/test_integration.py:954]: Can you move this up to the top of the function (after init)? That way we ensure each time it runs there aren't any leftover objects and also we could inspect what was loaded after it is done if there are any issues.

File: tests/core/engine_adapter/test_integration.py [hunk 6]
```diff
@@ -872,3 +881,74 @@ def test_scd_type_2(ctx: TestContext):
...
         ),
     )
+
+
+def test_sushi(ctx: TestContext):
+    if ctx.test_type == "df":
+        pytest.skip("Sushi end-to-end tests only need to run for query and pyspark tests")
+
+    # this skips the pyspark test if the engine isn't pyspark-enabled
+    ctx.init()
```
Comment (eakmanrq) [inline tests/core/engine_adapter/test_integration.py:891]: This is my bad that I coupled the pyspark check with init since you don't actually need the rest of init. I can clean that up in follow up PR and then allow this test to skip doing the full init when it doesn't need it.

File: tests/core/engine_adapter/test_integration.py [hunk 7]
```diff
@@ -872,3 +881,74 @@ def test_scd_type_2(ctx: TestContext):
...
+    context.plan(
+        environment="test_prod",
+        start=start,
+        end=end,
+        skip_tests=True,
+        no_prompts=True,
+        auto_apply=True,
+    )
+
+    data_validator = SushiDataValidator.from_context(context)
```
Comment (eakmanrq) [inline tests/core/engine_adapter/test_integration.py:914]: Thanks for including this! I look forward to when we have additional validations for other tables.

File: tests/core/engine_adapter/test_integration.py [hunk 8]
```diff
@@ -872,3 +881,74 @@ def test_scd_type_2(ctx: TestContext):
...
+    # Ensure that the plan has been applied successfully.
+    no_change_plan = context.plan(
+        environment="test_dev",
+        start=start,
+        end=end,
+        skip_tests=True,
+        no_prompts=True,
+        include_unmodified=True,
+    )
+    assert not no_change_plan.requires_backfill
```
Comment (eakmanrq) [inline tests/core/engine_adapter/test_integration.py:932]: Can also double check this is a new environment:
```
assert no_change_plan.context_diff.is_new_environment
```

File: tests/core/engine_adapter/test_integration.py (hunk 7 above)
Comment (eakmanrq) [reply tests/core/engine_adapter/test_integration.py:914]: That can be a later PR. I think making sure we validate the data for each model that has a unique load pattern would be good. I wonder if we should consider freezing in order to hard-code expected values. We can figure this out later though.
